blogs/allSettled.md

```diff
@@ -0,0 +1,47 @@
+<a href="https://github.com/kriskowal/q">Q</a> has <a href="https://github.com/kriskowal/q/wiki/API-Reference#promiseallsettled">Q.allSettled</a>.  <a href="https://docs.angularjs.org/api/ng/service/$q">$q</a> does not.  I've noticed a lot of AngularJS devs request $q.allSettled.  So here it is!  <a href="http://jsfiddle.net/0sr65m7t/">JS Fiddle</a>
+
+<!--more-->
+<pre>
+angular.module("qImproved", [])
+.config(function ($provide) {
+  $provide.decorator("$q", function ($delegate) {
+    function allSettled(promises) {
+      var deferred = $delegate.defer(),
+          counter = 0,
+          results = angular.isArray(promises) ? [] : {};
+
+      angular.forEach(promises, function(promise, key) {
+        counter++;
+        $delegate.when(promise).then(function(value) {
+          if (results.hasOwnProperty(key)) return;
+          results[key] = { status: "fulfilled", value: value };
+          if (!(--counter)) deferred.resolve(results);
+        }, function(reason) {
+          if (results.hasOwnProperty(key)) return;
+          results[key] = { status: "rejected", reason: reason };
+          if (!(--counter)) deferred.resolve(results);
+        });
+      });
+
+      if (counter === 0) {
+        deferred.resolve(results);
```

Comment:
$q.all is provided by default and does exactly what you describe.  I use $q.all a lot and think it's more useful than $q.allSettled.  But I'll call out the differences between $q.all and $q.allSettled.